frontend/src/queries/nodes/PersonsNode/PersonsSearch.tsx

```diff
@@ -4,38 +4,57 @@ import { Tooltip } from 'lib/lemon-ui/Tooltip'
 
 import { useDebouncedQuery } from '~/queries/hooks/useDebouncedQuery'
 import { PersonsNode, PersonsQuery } from '~/queries/schema'
+import { isInsightPersonsQuery, isPersonsQuery, isRetentionQuery } from '~/queries/utils'
 
 interface PersonSearchProps {
     query: PersonsNode | PersonsQuery
     setQuery?: (query: PersonsNode | PersonsQuery) => void
 }
 
+type actorType = 'person' | 'group'
+
+const labels: Record<actorType, any> = {
+    person: {
+        label: 'persons',
+        description:
+            'Search by email or Distinct ID. Email will match partially, for example: "@gmail.com". Distinct ID needs to match exactly.',
+    },
+    group: {
+        label: 'groups',
+        description:
+            'Search by group name or Distinct ID. Group name will match partially. Distinct ID needs to match exactly.',
+    },
+}
+
+function queryForGroup(query: PersonsNode | PersonsQuery): boolean {
+    return (
+        isPersonsQuery(query) &&
+        isInsightPersonsQuery(query.source) &&
+        isRetentionQuery(query.source.source) &&
+        query.source.source.aggregation_group_type_index !== undefined
```

Comment:
Not blocking for this PR, but this means we'll need to know to edit this specific file when adding new query types that also aggregate on groups.

```
        isRetentionQuery(query.source.source) &&
        query.source.source.aggregation_group_type_index !== undefined
```

To clean this a bit, this could probably get its own `isQueryWithGroupAggregation(query.source.source)` util.

Reply:
That's a bit tricky since a lot of type checking gets in the way (source is only present on specific queries and so on and so on). I moved the whole check to a util function.